### DISCUSSION TITLE
GPU OOM error

### DISCUSSION THREAD

**Discussion by ziwei-yin** [talmolab/sleap#1048] [Help!]
Hello community,

I am training sleap on a 13.30 minute video with 39700 frames, using a 3080 10Gb GPU. And I am able to do prediction on all frames using run training but not run inference. So I am wondering what is the obstacle that lead to OOM error.

![Screenshot 2022-11-14 155933](https://user-images.githubusercontent.com/56346611/201706627-f7a5ac28-0d94-433e-b084-d51ba34bcab6.png)

![image](https://user-images.githubusercontent.com/56346611/201706867-ed8a258a-890d-4145-9ba6-482afc86915f.png)

look forward to your help, and let me know if I can provide more information to diagnose this problem.

**Comment by roomrys** [comment]
Hi @JuliaY123,

Very sorry for the delayed response. I don't have an answer at the moment and am still looking into the problem. 

### Workarounds
- A possible workaround could be to run inference through the CLI using the [`sleap-track`](https://sleap.ai/guides/cli.html?highlight=sleap%20track#sleap-track) command instead of re-running training + inference. 
- Try the unet model instead of leap (leap uses a bit more memory)

### First Pass
Taking a first look at the problem there doesn't seem to be a reason why running inference after training should consume less memory than running inference alone. The only lead I can think of is: what if a different data provider is used `LabelsReader` vs `VideoReader` when running inference after training vs running just inference. Is this the case and if so, how would this affect memory usage? More updates to follow.

Thanks,
Liezl
